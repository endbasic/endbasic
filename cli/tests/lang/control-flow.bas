' EndBASIC
' Copyright 2020 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

i = 0

WHILE i < 5
    j = 0
    WHILE j < i
        PRINT i; j
        j = j + 1
    WEND
    i = i + 1
WEND

IF i = 5 THEN
    PRINT "Done!"
END IF

FOR i = 0 TO 3
    FOR j = 15 TO 10 STEP -2
        PRINT i; j
    NEXT
NEXT

IF i = 4 AND j = 9 THEN
    PRINT "Also done!"
END IF

WHILE TRUE
    i = 0
    @start
    PRINT i
    IF i = 5 THEN: GOTO @done: END IF
    i = i + 1
    GOTO @start
    @done: GOTO @out
WEND
@out: PRINT "Escaped infinite loop"
