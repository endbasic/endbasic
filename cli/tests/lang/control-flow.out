 1  0
 2  0
 2  1
 3  0
 3  1
 3  2
 4  0
 4  1
 4  2
 4  3
Done!
Done!
 0  15
 0  13
 0  11
 1  15
 1  13
 1  11
 2  15
 2  13
 2  11
 3  15
 3  13
 3  11
Also done!
Edges
Edges
Edges
Middle
Middle
Center
Middle
Middle
Edges
Edges
Edges
 0
 1
 2
 3
 4
 5
Escaped infinite loop
